SU2_CFD/src/numerics/NEMO/CNEMONumerics.cpp

```diff
@@ -348,285 +420,282 @@ void CNEMONumerics::GetViscousProjJacs(su2double *val_Mean_PrimVar,
                                        su2double **val_Jac_i, su2double **val_Jac_j,
                                        const CConfig *config) {
 
+  unsigned short iDim, iSpecies, jSpecies, iVar, jVar, kVar;
+  su2double rho, rho_i, rho_j, vel[3], T, Tve;
+  su2double mu, ktr, kve, *Ds, dij, Ru, RuSI;
+  su2double theta, thetax, thetay, thetaz;
```

Comment:
It would be relatively easy to optimize and save you the trouble of allocating things like dFdVi, dFdVj, etc.
This would also make writing the code a bit safer by not depending on so many member variables.

Step one: template this function on nVar and nSpecies (call it GetViscousProjJacs_impl)
Step two: move the various tmp fields into the function and allocate statically.
Step three: run the loops over the template NVAR and NSPECIES instead of the runtime vars nVar and nSpecies.
Step four: use this function as the instantiation point, basically:
```
switch (nVar) {
  case 4:
    switch (nSpecies) {
      case 1: return ComputeViscousJacobians<4, 1>(...);
      case 2: ....
      default: SU2_MPI::Error();
  case 5:
    switch (nSpecies) {

    }
  default: SU2_MPI::Error();
} 
```
I can also show you how to make a "dynamic" version that would be safe to call with any number of vars, species (instead of SU2_MPI::Error()), it is just a little bit more work.